### ISSUE TITLE
Training on higher density point clouds

### ISSUE THREAD

**Issue by JBKnights** [jac99/MinkLoc3D#8]
Hi,

I'm interested in training MinkLoc3D on a dataset with higher density point clouds such as MulRan or the data from the ICRA2022 GPR challenge, which I saw contained MinkLoc3D-based approaches for the first and second place approaches.  What hyperparameters (e.g. voxel size, learning rate, training epochs, batch size, etc.) were changed in order to train on the higher density inputs of these datasets?

Thanks,

Joshua Knights

**Comment by jac99** [comment]
Hi,
if you're looking for state of the art performance, I would recommend MinkLoc3Dv2: [link](https://github.com/jac99/MinkLoc3Dv2). The method has a slightly improved network architecture and is trained using a different loss function (truncated Smooth Average Precision) using very large batches (e.g. 2048 point clouds) - by employing staged backpropagation. This improved the results compared to MinkLoc3D.

For the ICRA GPR competition on high density GPR dataset I used the following configuration (for MinkLoc3Dv2 model):
-  Trained on a raw data (no ground plane removal; no downsampling)
- batch_size = 1536
- batch_split_size = 3
- Data augmentation: random rotation; small jitter and translation; 3D version of random erasing data augmentation
- Loss: Truncated SmoothAP with 4 positives per query and tau= 0.01
- Thresholds used when generating training sets: threshold for positives = 5 metres and threshold for negatives = 20 metres
- Voxel size: 0.8 deg, 0.15m, 0.15m (cylindrical coordinates)
- Global descriptor dimensionality: 256

Network architecture (number of layers and convolutional filters in each layer) and cylindrical coordinates and voxels sizes are set in the model_config file. I used the following model_config:
```
[MODEL]
model = MinkLoc
planes = 64,128,128,128,64
layers = 1,1,1,1,1
num_top_down = 2
conv0_kernel_size = 5
feature_size = 256
block=ECABasicBlock
pooling=GeM
coordinates=polar
quantization_step=0.8,0.15, 0.15

normalize_embeddings=False

```

**Comment by jac99** [comment]
For high-density point clouds, a good idea could be to downsample the point cloud-first. Winners of the first place in the GPR competition mentioned, that they downsampled the original point clouds and remove the points on the ground plane. This would speed up the training and inference significantly.

**Comment by JBKnights** [comment]
Thanks for the great and prompt responses!  It's been extremely helpful.
